player/configfiles.c

```diff
@@ -91,8 +92,10 @@ void mp_parse_cfgfiles(struct MPContext *mpctx)
 
     load_all_cfgfiles(mpctx, section, "mpv.conf|config");
 
-    if (encoding)
+    if (encoding) {
         m_config_set_profile(mpctx->mconfig, SECT_ENCODE, 0);
+        mp_input_enable_section(mpctx->input, "encode", MP_INPUT_EXCLUSIVE);
```

Comment:
I wanna change this to "encoding" as well so the profile and input section loaded are the same, but I can't find this documented anywhere so I don't know if that's a breaking change or not

Reply:
I can clarify this. :) We need to keep the old name here for backwards-compatibility.

The `{encode}` is a tag that tells mpv whether the keybind will work in encoding mode. By default, only q, escape and one more works in encoding mode.

If the user wants to bind more things, they need to manually add commands and tag each as `{encode}`.

You can see the default `{encode}` commands here:

https://raw.githubusercontent.com/mpv-player/mpv/master/etc/input.conf

However, I *would* be in favor of one thing. This:

```c
mp_input_enable_section(mpctx->input, "encode", MP_INPUT_EXCLUSIVE); // legacy
mp_input_enable_section(mpctx->input, "encoding", MP_INPUT_EXCLUSIVE); // mp 0.40+
```

That would have more consistency with the `[encoding]` profile section, which was a change introduced in 0.3.x here:

https://github.com/mpv-player/mpv/blob/master/DOCS/encoding.rst

Reply:
Of course if I change "encode" to "encoding" here, I'd also update the default config file. I was just curious if I'm expected to see users in the wild using this input section in their input.conf, because if that were to be the case then a breaking change for "consistency" is not worth it.

Reply:
I don't think renaming it is worthwhile.

Reply:
Oh and I noticed that `{encode}` is not documented anywhere in the codebase.

```
$ rg '\{encode\}'
etc/input.conf
84:#q {encode} quit 4
86:#ESC {encode} quit 4
164:#CLOSE_WIN {encode} quit 4
```

So technically it was never a public feature, and I doubt that users really set encoding keybinds. Perhaps renaming it to `{encoding}` right away without any backwards compat is fine.